hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/parse/UnsupportedOperationVisitor.java

```diff
@@ -275,6 +278,8 @@ public Void visit(SqlLiteral literal) {
                 // string literals to be of CHAR type, not VARCHAR. Validated type
                 // of string literals is still VARCHAR in HazelcastSqlValidator.
             case CHAR:
+            case TIMESTAMP:
+            case DATE:
```

Comment:
Why did we add TIMESTAMP and DATE, but not TIME and TIMESTAMP WITH TIME ZONE?

Reply:
I didn't change it right now because of two problems:
- Cannot create TimstampTZ literals because of the parser problem, therefore we cannot hit literal.visit stage for this type of values.
- I haven't fully implemented Time yet, I will add the case statement once it is ready.

Reply:
We should leave a TODO with the reason.